gateway/src/apicast/http_proxy.lua

```diff
@@ -101,26 +163,53 @@ local function forward_https_request(proxy_uri, proxy_auth, uri, skip_https_conn
         -- read and need to be cached in a local file. This request will return
         -- nil, so after this we need to read the temp file.
         -- https://github.com/openresty/lua-nginx-module#ngxreqget_body_data
-        body = ngx.req.get_body_data(),
-        proxy_uri = proxy_uri,
-        proxy_auth = proxy_auth
-    }
+        body = ngx.req.get_body_data()
+
+        if not body then
+            local temp_file_path = ngx.req.get_body_file()
+            ngx.log(ngx.INFO, "HTTPS Proxy: Request body is bigger than client_body_buffer_size, read the content from path='", temp_file_path, "'")
+
+            if temp_file_path then
+              body, err = file_reader(temp_file_path)
+              if err then
+                ngx.log(ngx.ERR, "HTTPS proxy: Failed to read temp body file, err: ", err)
+                ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
+              end
+
+              if is_chunked then
+                -- If the body is smaller than "client_boby_buffer_size" the Content-Length header is
+                -- set based on the size of the buffer. However, when the body is rendered to a file,
```

Comment:
> If the body is smaller than "client_boby_buffer_size" the Content-Length header is set based on the size of the buffer

Who is doing that? In other words, when all the conditions meet:
* the request is chunked,
* buffering is enabled
* the request body is small

Who sets the `Content-Length` header?

Reply:
The lua-resty-http will set the `Content-Length` based on the body that we passed in. But good catch I should have put more details in the comment

Reply:
Ok, I see,  It's because it is a string and the resty-http gets the length out of it. It happens [here](https://github.com/ledgetech/lua-resty-http/blob/v0.15/lib/resty/http.lua#L622-L627). I would make it explicit, but good enough.

Reply:
Yeah I agree because this is something that will come up again in future when troubleshooting but it doesn't need to be done in this PR, can be added at a later date that `if headers["Content-Length"]=nil then headers["Content-Length"]=#body` (this will at least be a useful reference for now)